backend/src/packages/sections/section.service.ts

```diff
@@ -70,6 +73,63 @@ class SectionService
     return section.toObject();
   }
 
+  public async update(
+    payload: SectionUpdateParametersDto & SectionUpdateRequestDto,
+  ): Promise<SectionGetAllItemResponseDto | null> {
+    const id = Number(payload.id);
+
+    const entity = await this.sectionRepository.find(id);
+
+    if (!entity) {
+      return null;
+    }
+
+    const section = entity.toObject();
+
+    if (section.type !== payload.type) {
+      null;
+    }
+
+    switch (payload.type) {
```

Comment:
Let's create a map (section type to update section method) with update specific section methods split out of main `update`, instead of using such kind of switch